src/pathfinder/yapf/yapf_ship_regions.cpp

```diff
@@ -40,23 +40,18 @@ inline uint ManhattanDistance(const CYapfRegionPatchNodeKey &a, const CYapfRegio
 
 /** Yapf Node for water regions. */
 template <class Tkey_>
-struct CYapfRegionNodeT {
+struct CYapfRegionNodeT : CYapfNodeT<Tkey_, CYapfRegionNodeT<Tkey_> > {
+	typedef CYapfNodeT<Tkey_, CYapfRegionNodeT<Tkey_> > base;
 	typedef Tkey_ Key;
 	typedef CYapfRegionNodeT<Tkey_> Node;
 
-	Tkey_       m_key;
-	Node       *m_hash_next;
-	Node       *m_parent;
-	int         m_cost;
-	int         m_estimate;
-
 	inline void Set(Node *parent, const WaterRegionPatchDesc &water_region_patch)
 	{
-		m_key.Set(water_region_patch);
-		m_hash_next = nullptr;
-		m_parent = parent;
-		m_cost = 0;
-		m_estimate = 0;
+		base::m_key.Set(water_region_patch);
+		base::m_hash_next = nullptr;
+		base::m_parent = parent;
+		base::m_cost = 0;
+		base::m_estimate = 0;
```

Comment:
This does not follow our coding style I believe (much of YAPF doesn't...)

Shouldn't it all just be `this->...` instead of `base::`

Reply:
IMO it's better to have all of YAPF being consistently incorrect, and then rename everything in one go. I might even volunteer for that one, but it definitely deserves its own PR.